docs/src/threadpool.rst

```diff
@@ -32,35 +55,146 @@ Data types
 
 .. c:type:: void (*uv_work_cb)(uv_work_t* req)
 
-    Callback passed to :c:func:`uv_queue_work` which will be run on the thread
-    pool.
+    Callback passed to :c:func:`uv_queue_work` which will be run on
+    the executor.
 
 .. c:type:: void (*uv_after_work_cb)(uv_work_t* req, int status)
 
     Callback passed to :c:func:`uv_queue_work` which will be called on the loop
-    thread after the work on the threadpool has been completed. If the work
-    was cancelled using :c:func:`uv_cancel` `status` will be ``UV_ECANCELED``.
+    thread after the work on the executor has been completed or cancelled.
+    If the work was cancelled using :c:func:`uv_cancel`,
+    then `status` will be ``UV_ECANCELED``.
+
+.. c:type:: uv_executor_t
+
+    Executor type. Use when overriding the default threadpool.
+
+.. c:type:: void (*uv_executor_init_func)(uv_executor_t* executor)
+
+    Called once, the first time :c:func:`uv_executor_queue_work` is called.
+    Do any initialization you want.
+
+.. c:type:: void (*uv_executor_destroy_func)(uv_executor_t* executor)
+
+    Called at some point ???
```

Comment:
I put a destroy routine in src/executor.c and wrapped it with UV_DESTRUCTOR. I think this uses atexit hooks?

Reply:
UV_DESTRUCTOR would use the dynamic linker/standard library functions to clean up, i.e. I think it would typically run after `atexit` (and possibly a lot earlier, if libuv is unloaded as a shared library)?